File: app/src/main/java/org/imozerov/streetartview/ui/observe/ArtListAdapter.java
package org.imozerov.streetartview.ui.observe;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.imozerov.streetartview.R;

import de.hdodenhof.circleimageview.CircleImageView;

/**
 * Created by imozerov on 05.02.16.
 */
public class ArtListAdapter extends RecyclerView.Adapter {

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View rootView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.art_object_in_list, parent, false);
        ViewHolder viewHolder = new ViewHolder(rootView);
        return viewHolder;

    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView author;
        public TextView artObjectName;
        public CircleImageView thumb;

        public ViewHolder(View itemView) {
            super(itemView);
            author = (TextView) itemView.findViewById(R.id.art_object_view_in_list_author);
            artObjectName = (TextView) itemView.findViewById(R.id.art_object_view_in_list_name);
            thumb = (CircleImageView) itemView.findViewById(R.id.art_object_view_in_list_thumb);
        }
    }


}

File: app/src/main/java/org/imozerov/streetartview/ui/explore/base/ArtListAdapter.kt
package org.imozerov.streetartview.ui.explore.base

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import org.imozerov.streetartview.R
import org.imozerov.streetartview.ui.extensions.loadImage
import org.imozerov.streetartview.ui.model.ArtObjectUi

/**
 * Created by imozerov on 05.02.16.
 */
class ArtListAdapter(private val context: Context, private var data: List<ArtObjectUi>?) : RecyclerView.Adapter<ArtListViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ArtListViewHolder {
        val rootView = LayoutInflater.from(parent.context).inflate(R.layout.art_object_in_list, parent, false)
        return ArtListViewHolder(this.context, rootView)
    }

    override fun onBindViewHolder(holder: ArtListViewHolder, position: Int) {
        val artObject = data!![position]
        if (holder.artObjectId != artObject.id) {
            holder.thumb.loadImage(artObject.thumbPicUrl)
        }
        holder.artObjectId = artObject.id
    }

    override fun getItemCount() = data!!.size

    override fun getItemId(position: Int) = data!![position].id.hashCode().toLong()

    fun setData(artObjectUis: List<ArtObjectUi>) {
        if (artObjectUis == data) {
            return
        }
        data = artObjectUis
        notifyDataSetChanged()
    }
}
